### PR TITLE
Change axis limits using scientific notation

### LINKED ISSUE #26869: Setting plot limits using scientific notation
Expected behavior
Having plotted a workspace/spectrum containing small numbers (<1E-4) I should be able to rescale it manually. For example I may want y=0 at the bottom.

Actual behavior
The plot settings dialog “Lower limit” and “Upper limit” only allows fixed point entry and a maximum of 4 decimal places. It is very reluctant to accept E or e for scientific notation and having done so, on re-entering the dialog it has re-rounded the number to fixed point.

Steps to reproduce the behavior
Generate a workspace with small Y numbers (mine is a suitably normalised Dark Counts run). Plot it (spectrum line plot or colour plot). Open the settings and try to rescale it.
To enter scientific notation it is necessary to ensure the content of the field is a valid number at all times. So type “1.5”, then move the cursor back and insert “e” and then “-” to get “1.e-5”. Alternatively pasting a fully formed number from elsewhere seems to work.

Platforms affected
Mantid 4.1, Workbench. Tried on Windows, probably all.

### PR DESCRIPTION
**Pull request by PhilColebrooke**
**Description of work.**
This pull request makes it possible to change a plot's axis using scientific notation in the figure options menu. This issue was originally reported by James Lord.

**To test:**
1. Plot a figure and open figure options.
2. In the Upper Limit and Lower Limit fields for both axes, check that you can input numbers in both decimal form (e.g. 2.55, -1.2, 3.14159265) and scientific notation (1e-3, 2.5e4).
3. Entering an invalid value such as "e", "-.", "5e." should cause an error dialog to appear.

Fixes #26869 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/python/mantidqt/widgets/plotconfigdialog/axestabwidget/__init__.py [hunk 1]
```diff
@@ -37,10 +37,15 @@ def from_ax_object(cls, ax):
     def from_view(cls, view):
         props = dict()
         props['title'] = view.get_title()
-        props['xlim'] = (view.get_xlower_limit(), view.get_xupper_limit())
+
+        try:
```
Comment (martyngigg) [inline qt/python/mantidqt/widgets/plotconfigdialog/axestabwidget/__init__.py]: Could you help me understand why the `try...except` is necessary? Doesn't the `QDoubleValidator` ensure that we always have a `float`?

File: qt/python/mantidqt/widgets/plotconfigdialog/axestabwidget/view.py [hunk 2]
```diff
@@ -55,10 +55,10 @@ def set_title(self, title):
 
     # X-Axis getters
     def get_xlower_limit(self):
-        return self.xlower_limit_spin_box.value()
+        return self.xlower_limit_line_edit.text()
```
Comment (martyngigg) [inline qt/python/mantidqt/widgets/plotconfigdialog/axestabwidget/view.py]: Should we convert to a `float` here? The `QDoubleValidator` should guarantee this is the case. The presenter can just assume it gets a float back then (similarly for the other limits)

**Review by martyngigg — CHANGES_REQUESTED** [review]
This is much better with a free-form text box.

Just a couple of questions regarding the implementation.

**Comment by martyngigg** [comment]
@PhilColebrooke Have you had a chance to look at the comments here?

**Comment by PhilColebrooke** [comment]
> @PhilColebrooke Have you had a chance to look at the comments here?

Not yet.

**Comment by PhilColebrooke** [comment]
@martyngigg The ```try...except``` was there because in tests the view was mocked and so it was trying to run ```float()``` on a mock which was causing an error. I've removed the ```try...except``` and changed the tests so that the getters for the axis limits return a number so ```float()``` can be used. The values in the axis limit line edits still need to be converted to floats because my understanding is that ```QDoubleValidator``` just makes it so that the user can only input things which can be converted to a float, but the line edits themselves still return a string.

**Comment by martyngigg** [comment]
@PhilColebrooke Thanks for the explanation. That makes sense. Is it possible for the view `get_xlower_limit` (and like members) to do the float conversion? If the double validator is present it is guaranteed to be a double so when returning the view could do the conversion and then the client code doesn't have to care?

**Comment by PhilColebrooke** [comment]
@martyngigg The view now does the float conversion and everything seems to be working fine. This also meant that the tests don't have to be amended anymore which is nice.

**Review by martyngigg — APPROVED** [review]
Good stuff. This looks good now. :+1:

**Pull request merged**